src/trousse/dataset.py

```diff
@@ -314,38 +314,35 @@ def many_nan_columns(self) -> Set[str]:
         return many_nan_columns
 
     @property
-    def same_value_cols(self) -> Set[str]:
-        """
-        Return name of the columns containing only one repeated value.
+    def constant_cols(self) -> Set[str]:
+        """Return name of the columns containing only one repeated value.
 
         Returns
         -------
         Set[str]
             Set of column names with only one repeated value
         """
-        same_value_columns = set()
-        for c in self.df.columns:
-            # Check number of unique values
-            if len(self.df[c].unique()) == 1:
-                same_value_columns.add(c)
+        df_constant = self.df.loc[
+            :, self.df.apply(pd.Series.nunique, dropna=False) == 1
+        ]
```

Comment:
What about writing like this?
```suggestion
        df_nunique = self.df.nunique()
        constant_cols = df_nunique[df_nunique == 1].index
        return set(constant_cols)
```
Is the apply faster? (I find this solution a bit more straight-forward https://thispointer.com/pandas-get-unique-values-in-single-or-multiple-columns-of-a-dataframe-in-python/ )

Reply:
In that case you should remove the next `return` (since there was a deleted line in the middle, I could not include that in the suggestion :( )